extensions/amp-date-picker/0.1/amp-date-picker.js

```diff
@@ -503,49 +523,75 @@ class AmpDatePicker extends AMP.BaseElement {
   /**
    * Returns `true` if an info template exists.
    * @return {boolean}
+   * @private
    */
   hasInfoTemplate_() {
     return this.templates_.hasTemplate(this.element, '[info-template]');
   }
 
-  /** @return {!Promise<string>} */
+  /**
+   * Render any template for the info section of the date picker.
+   * @return {!Promise<string>}
+   * @private
+   */
   renderInfoTemplate_() {
     const template = this.element.querySelector('[info-template]');
-    return this.renderTemplate_(template);
+    if (template) {
+      return this.renderTemplateElement_(template)
+          .then(element => {
+            element.classList.add('i-amphtml-amp-date-picker-info');
+            return this.getRenderedTemplateString_(element);
+          });
+    } else {
+      return Promise.resolve('');
+    }
   }
 
   /**
-   * Render the given template with the given data. If the template does not
-   * exist, use a fallback string.
+   * Render the given template into an element with the given data.
+   * @param {!Element} template
+   * @param {!JsonObject=} opt_data
+   * @return {!Promise<!Element>}
+   * @private
+   */
+  renderTemplateElement_(template, opt_data = /** @type {!JsonObject} */ ({})) {
+    return this.templates_.renderTemplate(template, opt_data);
+  }
+
+  /**
+   * Render the given template into text with the given data.
    * The fallback string will be rendered directly into the DOM so it must
    * not contain unsanitized user-supplied values.
    * @param {?Element} template
    * @param {!JsonObject=} opt_data
    * @param {string=} opt_fallback
    * @return {!Promise<string>}
+   * @private
    */
   renderTemplate_(template, opt_data, opt_fallback = '') {
     if (template) {
-      const data = opt_data || /** @type {!JsonObject} */ ({});
-      return this.templates_.renderTemplate(template, data)
-          .then(rendered => {
-            const renderedEvent = createCustomEvent(
-                this.win_,
-                AmpEvents.DOM_UPDATE,
-                /* detail */ null,
-                {bubbles: true});
-            this.container_.dispatchEvent(renderedEvent);
-            return rendered./*REVIEW*/outerHTML;
-          });
+      return this.renderTemplateElement_(template, opt_data)
+          .then(rendered => this.getRenderedTemplateString_(rendered));
     } else {
       return Promise.resolve(sanitizeFormattingHtml(opt_fallback));
     }
   }
 
+  /**
+   * Convert a rendered template element to a string
+   * @param {!Element} rendered
+   * @return {string}
+   * @private
+   */
+  getRenderedTemplateString_(rendered) {
```

Comment:
Why are we not okay with outerHTML and why is it okay here?

Reply:
Assigning to it can be dangerous, and reading from it can be expensive. But since it's only read once per template, it's ok here.